### ISSUE TITLE
a11y test verification from Accessibility

### ISSUE THREAD

**Issue by humancompanion-usds** [department-of-veterans-affairs/va.gov-team#55907]
## What does your team need support for? Check all that apply.
- [ ] Launching one or more unauthenticated pages, online forms, apps, or tools on VA.gov
- [ ] Making revisions to one or more existing VA.gov pages
- [x] Something else 
 
 *Give a brief description of what your team needs support with.*
The Design System Team needs as assist from a11y on this issue:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/26274

The issue is a bit vague and thus we need someone to re-test to see if it's even an issue we need to solve:
> The alert box is not being announced to JAWS or NVDA on Chrome b/c of the role="alert" on the error messages. Is this a current standard behavior on the forms? Can we manage focus and set it on the Alertbox, or is that on teams to manage outside this component?

## Will this new product be released incrementally (for instance 25% of users initially)? 
  - [ ] Yes
  - [x] No
 
**Note:** If you checked yes, we'll reach out to discuss details about the content in the react widget.
  
## When do you expect to launch your product to 100% of users?

If the va-alert web-component does indeed need a fix then we'll prioritize fixing that bug in a future sprint and once we do fix it then we'll deploy that fix immediately to 100% of users.  

*Please provide an estimated date so our team can create other relevant tickets.*

Target would be sprint 27 starting May 11, 2023. 
 
## Supporting artifacts
*Please provide supporting artifacts as available.*

All I have is this old issue. If we can close it as invalid then great. If not, we need details on how it fails.

## Will this work be going through the Collaboration Cycle?
- [ ] Yes 
- [x] No

## When does this work need to be done?
- Estimated launch date: 
- Estimated staging review date:
- Content, Accessibility, and IA work needed by: May 11, 2023

## Do you plan to bring this to an upcoming content office hours session?
- [ ] Yes, benefit content office hours (Thursdays, 3:00 p.m. to 3:45 p.m. ET)
- [ ] Yes, health  content office hours (Thursdays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] Yes, unauth office hours (Mondays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] No, but I'd like to schedule time to talk about this request
- [x] No, let's work asynchronously and meet if needed

**Note:** If we think this work would benefit from a collaborative session with you, we may ask you to bring it to office hours or set up a separate time to meet.

## About your team

- Team name: Design System
-	OCTO-DE product owner: Matthew Dingee
-	Product manager: Carol Wong
-	Designer: Natalie Hill
-	FE engineer: Jami Gibbs
-	Product/team Slack channel: #platform-design-system

## Next steps

**Once you’ve submitted this ticket, please post a link to this issue in the [#sitewide-content-ia](https://dsva.slack.com/channels/sitewide-content-ia) Slack channel and tag Randi Hecht.**

If you also need engineering support from the Public Websites team, fill out their [intake request form](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=Public+Websites%2C+Sitewide+content&labels=vsa-public-websites%2C+vsa%2C+vsa-public-websites-intake%2C+sitewide-content%2C+needs-grooming&template=public-websites-intake.md&title=%3CType+of+Request%3E+from+%3CTeam%3E).
  
If you need a page/URL redirected, a URL changed or a vanity URL set up, please submit a [Redirect, URL change, or vanity URL request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=mnorthuis&labels=ia&template=redirect-request.md&title=Redirect+Request)

**Comment by joshkimux** [comment]
[This ticket from supplemental claims has some testing that may be relevant to this](https://github.com/department-of-veterans-affairs/va.gov-team/issues/53446#issuecomment-1456200388)

**Comment by SarahKay8** [comment]
Hey @humancompanion-usds! I currently am without a Windows (have been for about a month) at the moment and can't run Jaws and NVDA on a MAC. I'm in communication with folks to get I and Sara Smith a second laptop in order to test this out for you

**Comment by humancompanion-usds** [comment]
Thanks, @SarahKay8! Much appreciated. Let me know how it goes.

**Comment by humancompanion-usds** [comment]
Hey there - Can I get an update on this? Looks like it perhaps is in progress now?

**Comment by SarahKay8** [comment]
Hey Matt! Got my windows laptop. Can have this wrapped up before end of sprint

**Comment by SarahKay8** [comment]
@humancompanion-usds When we have role=alert on it and we press save it still doesn’t announce out on chrome + Jaws. Our best guess is it might have to deal with user react components here. The react components are being added into the page so it is not working in the way role=alert is supposed to, what’s happening with the react component is the alert which has role=alert is being loaded into the page and not on page load which is probably creating this bug. 
What we really want to do is have an existing live region on your page, such as
<div role="alert" id="someID"></div>
and then the <span> that is returned to you from the server does not need role="alert". It can just return a <span> with the error message contained within it
<span>this is an alert</span>
and then you have to inject this message into the <div>
<div role="alert" id="someID">
  <span>this is an alert</span>
</div>


Discovery work may be necessary for this ticket.

**Comment by humancompanion-usds** [comment]
Hmm. That sounds like perhaps the very first bullet of our accessibility considerations guidance for Alert is incorrect? https://design.va.gov/components/alert#accessibility-considerations

However, this sounds like an implementation detail in how the Alert component is used rather than a problem with the component itself. I'll assign the 26274 issue to an engineer on the Design System Team to get to the bottom of this. Thanks for the investigation.

**Comment by humancompanion-usds** [comment]
Sorry, @SarahKay8 - One point of clarification:

> When we have role=alert on it and we press save 

I'm confused. When `role=alert` is an attribute of an Alert component why is the user pressing save (and where)? Also, Alerts should not be dynamically added to a page in a form flow. The page should be reloading and displaying the Alert. Is there a test page or application that you tested this in that the Design System Team can reference?

**Comment by SarahKay8** [comment]
Hi Matt so I was referring to the form on supplemental claims. After pressing save when editing a home phone number, the alert isn't being read via Jaws+ Chrome. https://images.zenhubusercontent.com/133843125/ae293c83-40ca-4d9d-bb42-9b38024f1d90/224691508_0763cb14_8c02_4348_b5c6_a3d32e4a8d36.mov
